tests/rados/test_election_strategies.py

```diff
@@ -222,3 +283,382 @@ def check_mon_status(rados_obj, mon_obj):
             return False
         log.info(f"Mon {mon} is in healthy state: {status} | {desc}")
     return True
+
+
+def add_all_mons_to_disallow_list(mon_obj) -> None:
+    """Method to validate last Mon cannot be added to disallowed leaders list
+
+    Steps:
+    - Get list of all Mons from the cluster
+    - Add all Mons to disallowed leaders list
+    - Validate last allowed Mon cannot be added to disallow list
+    - Revert cluster back to original state by removing all Mons from disallow list
+
+    Args:
+        mon_obj: object MonElectionStrategies
+    Returns:
+        Raises Exception when test fails
+    Example:
+
+    """
+    log.info(
+        "\n -------------------------------------------"
+        "\n Starting validation:"
+        "\n `last allowed Mon` cannot be added to disallow list"
+        "\n -------------------------------------------"
+    )
+
+    mons = list(mon_obj.get_mon_quorum().keys())
+    last_allowed_mon = mons.pop()
+
+    for mon in mons:
+        log.info(f"Adding mon {mon} to disallow list")
+        if not mon_obj.set_disallow_mon(mon=mon):
+            raise AssertionError(f"could not add Mon {mon} to disallowed list")
+        log.info(f"Successfully added mon {mon} to disallow list")
+
+    log.info(
+        f"Adding last remaining allowed mon {last_allowed_mon} to disallow list\n"
+        "Proceeding to validate error generation when last allowed mon is added to disallow list"
+    )
+    try:
+        if mon_obj.set_disallow_mon(mon=last_allowed_mon):
+            raise AssertionError(
+                f"Last allowed mon {last_allowed_mon} added to disallow list"
+            )
+
+    except AssertionError as ae:
+        raise AssertionError(ae.args[0])
+
+    except Exception as e:
+        expected_error_msg = f"Error EINVAL: mon.{last_allowed_mon} is the only remaining allowed leader!"
+        generated_error_msg = e.args[0].split("\n")[4]
+        if generated_error_msg != expected_error_msg:
+            raise AssertionError(
+                f"Generated Error: {generated_error_msg}"
+                f"Expected Error: {expected_error_msg}"
+            )
+        log.info(
+            f"Validated Mon {last_allowed_mon} could not be added to diallow list\n"
+            f"Error generated: {e.args}\n"
+        )
+
+    log.info(
+        "Successfully validated Last remaining Mon cannot be added to disallow list\n"
+        "Reverting cluster back to original state\n"
+        f"Proceeding to remove mons {mons} from disallow list"
+    )
+
+    for mon in mons:
+        log.info(f"Removing mon {mon} from disallow list")
+        if not mon_obj.remove_disallow_mon(mon=mon):
```

Comment:
same as above.

Reply:
Added pretty_print=True for remove_disallow_mon()

https://github.com/red-hat-storage/cephci/blob/83e34286c4e52f3067a54ba0b8b151b8345e9bde/tests/rados/monitor_configurations.py#L60-L76